### PR TITLE
19179 Added rest of amalgamating businesses unit tests

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#19179

*Description of changes:*
    - app version = 5.6.41
    - added rest of unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: tests/unit/AmalgamatingBusinesses.spec.ts [hunk 1]
```diff
@@ -24,7 +27,8 @@ describe('Amalgamating Businesses - components and validity', () => {
 
   it('renders as staff', async () => {
     // set state
-    await store.setKeycloakRoles(['staff'])
+    store.setKeycloakRoles(['staff'])
+    await Vue.nextTick()
```
Comment (severinbeauvais) [inline tests/unit/AmalgamatingBusinesses.spec.ts:31]: Awaiting the store action worked but VS Code said "await has no effect on the type of this expression" which is what you'd see if setKeycloakRoles wasn't async.

So I did this instead, which also works and doesn't have a VS Code message that someone might want to optimize (remove) and thus break the test.

File: tests/unit/AmalgamatingBusinesses.spec.ts [hunk 2]
```diff
@@ -87,15 +93,368 @@ describe.skip('Amalgamating Businesses - add amalgamating business', () => {
...
-  it('saves an amalgamating business', async () => {
+  it('save an amalgamating business - BC', async () => {
+    // set state
+    store.setCurrentDate('2023-01-01')
+    await Vue.nextTick()
+
+    // open panel
+    await wrapper.setData({ isAddingAmalgamatingBusiness: true })
+
+    // mock services functions
```
Comment (severinbeauvais) [inline tests/unit/AmalgamatingBusinesses.spec.ts:338]: At least look at this section 😆 

Spying on service functions and mocking their implementation (or result) is super handy.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-627-kaf6v679.web.app

File: tests/unit/AmalgamatingBusinesses.spec.ts [hunk 3]
```diff
@@ -87,15 +93,368 @@ describe.skip('Amalgamating Businesses - add amalgamating business', () => {
...
+    expect(business.name).toBe('My BC Business')
+    expect(business.email).toBe('email@example.com')
+    expect(business.legalType).toBe(CorpTypeCd.BC_COMPANY)
+    expect(business.address.streetAddress).toBe('123 Main St')
+    expect(business.isNotInGoodStanding).toBe(false)
+    expect(business.isFrozen).toBe(false)
+    expect(business.isFutureEffective).toBe(false)
+    expect(business.isPendingFiling).toBe(false)
+    expect(business.isLimitedRestoration).toBe(false)
+    expect(business.isHistorical).toBe(false)
```
Comment (severinbeauvais) [inline tests/unit/AmalgamatingBusinesses.spec.ts:388]: In retrospect, I could have checked if `business` was "equal to" an object with all these properties.

File: tests/unit/AmalgamatingBusinesses.spec.ts (hunk 2 above)
Comment (ketaki-deodhar) [reply tests/unit/AmalgamatingBusinesses.spec.ts:338]: Good learning here!

**Review by ketaki-deodhar — APPROVED** [review]
LGTM!

File: tests/unit/AmalgamatingBusinesses.spec.ts [hunk 4]
```diff
@@ -87,15 +93,368 @@ describe.skip('Amalgamating Businesses - add amalgamating business', () => {
...
+
+    // simulate saving a BC business
+    await wrapper.vm.saveAmalgamatingBusiness({
+      legalType: CorpTypeCd.BC_COMPANY,
+      name: 'My BC Business',
+      identifier: 'BC1234567'
+    })
+
+    // verify data
+    expect(store.stateModel.amalgamation.amalgamatingBusinesses.length).toBe(1)
```
Comment (JazzarKarim) [inline tests/unit/AmalgamatingBusinesses.spec.ts:168]: Can't we use the store getter (`getAmalgamatingBusinesses`) here and below instead Sev?

File: tests/unit/AmalgamatingBusinesses.spec.ts (hunk 2 above)
Comment (JazzarKarim) [reply tests/unit/AmalgamatingBusinesses.spec.ts:338]: Love it.

**Review by JazzarKarim — APPROVED** [review]
Looks great!

File: tests/unit/AmalgamatingBusinesses.spec.ts (hunk 4 above)
Comment (severinbeauvais) [reply tests/unit/AmalgamatingBusinesses.spec.ts:168]: I think so. Want me to change it?

Comment (JazzarKarim) [reply tests/unit/AmalgamatingBusinesses.spec.ts:168]: No need, was just curious. Resolving this conversation Sev.

**Pull request merged**